sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -2320,6 +2320,107 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSQLContext {
     assert(ex4.getMessage.contains("cannot resolve '`a`'"))
   }
 
+  test("forall function - array for primitive type not containing null") {
```

Comment:
I think most/all of this should be covered by unit test. You can add a single test to validate that the function registry works, if you must. I know you mirrored the tests for `exists` and I think they have the same problem. In general we should test the interface here (including the errors), and not so much the underlying functionality (that should be covered by UTs).

Reply:
I agree with it, but I remember there was a problem that the behavior was different between the whole stage codegen on/off before, then we decided to do like this as a workaround (#21795).